src/security/cert_generators/file/security_file_certgen.cc

```diff
@@ -276,29 +292,38 @@ int main(int argc, char *argv[])
             }
         }
 
+        // when -s is used, -M is required
+        if (!db_path.empty() && max_db_size == 0)
+            throw std::runtime_error("security_file_certgen is missing the required parameter. There should be -s and -M parameters together, or neither used.");
```

Comment:
Please consider this simpler version:

```
"security_file_certgen -s requires a positive -M parameter"
```

As discussed in another comment, there will be another check here that says

```
"security_file_certgen -M requires a -s parameter"
```

If you want to combine the two checks, here is a simple version:

```
const bool useDb = !db_path.empty();
...
if (!(useDb == (max_db_size > 0)))
    throw std::runtime_error("security_file_certgen -s and -M parameters must be given or omitted together");
```

Reply:
The -M parameter does not have anything to do with positive or negatives.

IMO the check you propose is depending on language details too subtle to be called a "simple version". It is the form of check I try to only use in high performance code paths where performance needs outweigh simplicity. Helper startup is not that desperate for CPU cycles.